changelog.d/12808.feature

```diff
@@ -0,0 +1 @@
+Update to `check_event_for_spam`. Deprecate the current callback signature, replace it with a new signature that is both less ambiguous (replacing booleans with explicit allow/block) and more powerful (ability to return explicit error codes).
```

Comment:
The nature of the change means we need a paragraph in the upgrade notes.

Reply:
I... don't follow.